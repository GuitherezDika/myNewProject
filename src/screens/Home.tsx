import React, { useCallback, useEffect } from 'react';
import { SafeAreaView } from 'react-native';
import { styles } from '../styles';
import { useFocusEffect } from '@react-navigation/native';
import { ImageList } from '../components/ImageList';
import { useTheme } from '../components/theme/ThemeContext';

function HomeScreen() {
  const { theme } = useTheme();

  useEffect(() => {
    // hanya di panggil 1x saat masuk ke screen, dan tidak akan ke hit walaupun sudah bolak balik di klik
    console.log('home screen');
  }, []);

  useFocusEffect(
    // Akan run setiap button home di klik
    useCallback(() => {
      console.log('callback HOME ^^^^', theme);
    }, [])
  )

  return (
    <SafeAreaView style={[styles.container, { backgroundColor: theme.background}]}>
      <ImageList />
    </SafeAreaView>
  );
}

export default HomeScreen;