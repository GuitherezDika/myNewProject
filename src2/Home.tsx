import { Text, TextInput, View } from "react-native"
import React, { useState } from "react";

const Home = () => {
    const [data, setData] = useState('test');
    const onChange = (x: string) => {
        setData(x);
    };

    return (
        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
            <TextInput
                testID="username"
                onChangeText={text => onChange(text)}
                style={{ backgroundColor: 'gray', marginBottom: 15 }}
                placeholder="Enter the username"
            />
        </View>
    )
}

export default Home;