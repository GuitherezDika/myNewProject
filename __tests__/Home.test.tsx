import 'react-native';
import React from 'react';
import Home from '../src2/Home';
import renderer from 'react-test-renderer';
import {it, test, expect} from '@jest/globals';

test('Home snapShot', () => {
    const snap = renderer.create(
        <Home />
    ).toJSON();
    expect(snap).toMatchSnapshot();
})  