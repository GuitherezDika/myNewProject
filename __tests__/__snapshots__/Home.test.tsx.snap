// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter the username"
    style={
      {
        "backgroundColor": "gray",
        "marginBottom": 15,
      }
    }
    testID="username"
  />
</View>
`;
